File: Ruby/Civitas/pkg/Civitas-0.0.1/lib/controlador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

# encoding: utf-8
require_relative 'civitas_juego'
require_relative 'vista_textual'
require_relative 'operaciones_juego'
require_relative 'respuestas'

module Civitas
  class Controlador
    def initialize(juego, vista)
      @juego=juego
      @vista=vista
    end
    
    attr_accessor :juego, :vista
    
    def juega
      # Para ahorrar cantidad de escritura
      comprar=Operaciones_juego::COMPRAR
      gestionar=Operaciones_juego::GESTIONAR
      salir_carcel=Operaciones_juego::SALIR_CARCEL
      pasar_turno=Operaciones_juego::PASAR_TURNO
      si=Respuestas::SI
      
      @vista.set_civitas_juego(juego)
      # print "\nBIENVENIDO A CIVITAS!\n"
      @vista.pausa
      fin=@juego.final_del_juego
      
      while(!fin)
        @vista.actualizar_vista
        
        siguiente=@juego.siguiente_paso
        @vista.mostrar_siguiente_operacion(siguiente)
        
        if(siguiente!=pasar_turno)
          @vista.mostrar_eventos
        end
        
        if(!fin)
          if(siguiente==comprar)
              respuestas=@vista.comprar
            if(respuestas==si)
              @juego.comprar
              @juego.siguiente_paso_completo(comprar)
            else
              @juego.siguiente_paso_completo(comprar)
            end
          elsif(siguiente==gestionar)
             @vista.gestionar
             op=OperacionInmobiliaria.new(@vista.get_gestion, @vista.get_propiedad)
             
             if(op.get_gestion==Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA)
               @juego.cancelar_hipoteca(@vista.get_propiedad)

             elsif(op.get_gestion==Civitas::GestionesInmobiliarias::CONSTRUIR_CASA)
               @juego.construir_casa(@vista.get_propiedad)

             elsif(op.get_gestion==Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL)
               @juego.construir_hotel(@vista.get_propiedad)

             elsif(op.get_gestion==Civitas::GestionesInmobiliarias::HIPOTECAR)
               @juego.hipotecar(@vista.get_propiedad)

             elsif(op.get_gestion==Civitas::GestionesInmobiliarias::VENDER)
               @juego.vender(@vista.get_propiedad)

             elsif(op.get_gestion==Civitas::GestionesInmobiliarias::TERMINAR)
               @juego.siguiente_paso_completo(gestionar)
             end
          elsif(siguiente==salir_carcel)
              salida=@vista.salir_carcel
            if(salida==Civitas::SalidasCarcel::PAGANDO) 
              @juego.salir_carcel_pagando
            else
              @juego.salir_carcel_tirando
            end
            
            @juego.siguiente_paso_completo(salir_carcel)
          end
        end
      end
      
      ranking=@juego.ranking
      puts "\n\t-CLASIFICACION FINAL-\n"
      
      for i in(0..rank.length)
        puts "#{i+1} rank. #{ranking.get(i).get_nombre}"
      end
    end
  end
end

File: Java/Civitas/src/GUI/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GUI;
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import civitas.civitasJuego;
import civitas.Operacioninmobiliaria;
import civitas.OperacionesJuego;
import civitas.Gestionesinmobiliarias;
import civitas.Jugador;
import civitas.SalidasCarcel;
import java.util.ArrayList;

/**
 *
 * @author alonsoarturo
 */
public class Controlador{
    private final civitasJuego juego;
    private final CivitasView vista;
    
    Controlador(civitasJuego juego, CivitasView vista){
        this.juego=juego;
        this.vista=vista;
    }
    
    void juega(){
        vista.setCivitasJuego(juego);
        //System.out.print("\nBIENVENIDOS A CIVITAS:\n\n");
        
        boolean fin=juego.finalDelJuego();
        
        while(fin==false){
            vista.actualizarVista();
            
            OperacionesJuego siguiente=juego.siguientePaso();
            vista.mostrarSiguienteOperacion(siguiente);
            
            if(siguiente!=OperacionesJuego.PASAR_TURNO){
                vista.mostrarEventos();
            }
            
            if(fin==false){
                if(siguiente==OperacionesJuego.COMPRAR){
                        Respuestas respuesta=vista.Comprar();
                    if(respuesta==Respuestas.SI){
                        juego.comprar();
                        juego.siguientePasoCompleto(OperacionesJuego.COMPRAR); 
                    }else{
                        juego.siguientePasoCompleto(OperacionesJuego.COMPRAR);
                    }
                }else{
                    if(siguiente==OperacionesJuego.GESTIONAR){
                        vista.gestionar();
                        
                        Operacioninmobiliaria op=new Operacioninmobiliaria(Gestionesinmobiliarias.values()[vista.getGestion()], vista.getPropiedad());                   
                        
                        if(op.getGestion()==Gestionesinmobiliarias.CANCELAR_HIPOTECA){
                            juego.cancelarHipoteca(vista.getPropiedad());
                        }else{
                            if(op.getGestion()==Gestionesinmobiliarias.CONSTRUIR_CASA){
                                juego.construirCasa(vista.getPropiedad());
                            }else{
                                if(op.getGestion()==Gestionesinmobiliarias.CONSTRUIR_HOTEL){
                                    juego.construirHotel(vista.getPropiedad());
                                }else{
                                    if(op.getGestion()==Gestionesinmobiliarias.HIPOTECAR){
                                        juego.hipotecar(vista.getPropiedad());
                                    }else{
                                        if(op.getGestion()==Gestionesinmobiliarias.VENDER){
                                            juego.vender(vista.getPropiedad());
                                        }else{
                                            if(op.getGestion()==Gestionesinmobiliarias.TERMINAR){
                                                juego.siguientePasoCompleto(OperacionesJuego.GESTIONAR);
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }else{
                        if(siguiente==OperacionesJuego.SALIR_CARCEL){
                                SalidasCarcel salida=vista.salirCarcel();
                            if(salida==SalidasCarcel.PAGANDO){
                                juego.salirCarcelPagando();
                            }else{
                                juego.salirCarcelTirando();
                            }
                            juego.siguientePasoCompleto(OperacionesJuego.SALIR_CARCEL);
                        }
                    }
                }
            }
        }
    ArrayList<Jugador> ranking=juego.ranking();
    System.out.print("RANKING");
    
    for(int i=0; i<ranking.size(); i++){
        System.out.println(i+1);            
        System.out.println(ranking.get(i).toString());
    }
    
    }
}




